VogtInterface/ProductsBase.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtInterface
+{
+    abstract class ProductsBase: IProductsModel
+    {
+        public string Name { get; set; }
+        public decimal Price { get; set; }
+
+        public void PrintProductDetails()
+        {
```

Comment:
Awesome exactly!

So here you're using both polymorphism and an interface. Polymorphism for a protected `PrintProductSpecificDetails` function (which others cant see), but an interface to define the contract that other classes can see (specifically, the `PrintProductDetails` field).

Two concepts being used to provide a cleaner, encapsulated print function.